wajig.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH wajig 1 "March, 2012"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
wajig \- Simplified command line administrator for Debian
.SH SYNOPSIS
.B wajig
.RI [ options ] " commands" ...
.SH DESCRIPTION
This manual page briefly documents the
.B wajig
command.
.PP
.\" TeX users may be more comfortable with the \fB<whatever>\fP and
.\" \fI<whatever>\fP escape sequences to invode bold face and italics, 
.\" respectively.
\fBwajig\fP packages into one tool many commands useful for managing a
Debian system. Instead of having to remember whether to use dpkg or
apt-get or apt-cache, etc, wajig does the selection of the appropriate
tool for you.
.PP
\fBwajig\fP is a user command but will use \fBsudo\fP to run commands
requiring super user permissions.
.PP
The primary documentation is available from https://wajig.togaware.com.

.SH OPTIONS
This program follows the usual GNU command line syntax, with long
options starting with two dashes (`-').
A summary of options to get you started is included below.
For more details see \fBwajig tutorial\fP and \fBwajig commands\fP.
.TP
.B \-h, \-\-help
Show summary.
.TP
.B \-V, \-\-version
Show version of program.
.SH AUTHOR
This manual page was written by Graham Williams <Graham.Williams@togaware.com>,
for the Debian GNU/Linux system (but may be used by others).


NEWS
Wajig is now a native Debian package.
See /usr/share/doc/wajig/changelog for NEWS.

wajig (2.0.22) unstable; urgency=low

  * New command LISTSECTION and LISTSECTIONS.

wajig (2.0.21) 2005/02/01

    - Exit with status 1 if LISTNAMES returns no results. Bug reported by 
      A Costa (Closes: #292581).

wajig (2.0.20) 2005/01/29

    - Take over Debian maintenance from Dirk. Thanks for all your efforts
      Dirk!
    - Fix a locale problem where LC_ALL was being set to C by wajig for 
      all external system calls. Only needed for STATUS? Bug reported
      by Serge Matveev (Closes: #288852).
    - Fix bug in getting username when USER not define, reported by
      Dirk Eddelbuettel
    - Fix typo reported by A Costa (Closes: #292456)
    - LISTALTS should ignore the README Reported by A Costa 
      (Closes: #292457)
    - Fix change of DEPENDEES to DEPENDENTS Reported by A Costa 
      (Closes: #292469)

wajig (2.0.19) 2005/01/07

    - Fix problem when using sudo, and running wajig as root - if root
      was not in sudoers file then wajig could not run. root did not need
      to be in the sudoers file to run wajig - added a check for root 
      running wajig and if so, don't use sudo if sudo is installed.
      Bug reported by Serge Matveev (Closes: #288849).
    - Fix bug with PURGE-ORPHANS whereby sudo was being doubly used
      and causing problems if root was not in sudoers. Bug reported by
      Serge Matveev (Closes: #288847)
    - Continue work on command interpreter completion. Only command 
      complete the first "word" not all "words" in the command line.
    - REC-DOWNLOAD has been renamed RECURSIVE (but kept for
      compatibility)

wajig (2.0.18) 2005/01/03

    - Some simple documentation changes.
    - Without any arguments wajig will start a command interpreter.
      The command interpreter accepts any wajig command, with completion.
      Suggested by Lorenzo Prince (Closes: #288041).
    - Fix small bug in WHICHPKG which generated blank lines in output.

wajig (2.0.17) 2004/12/30

    - DETAILS now allows DEB files as arguments. Calls dpkg-deb --info.
    - CHANGELOG now tries harder to get most recent changelog. Rather
      than relying on http://changelogs.debian.net/<pkg>, use http://
      packages.debian.org/changelogs/pool/..../<pkg>_<version>/changelog.
      Delay in the updating of changelogs.debian.net links pointed out
      by Matthew Hawkins (and Closes: #286580).

wajig (2.0.16) 2004/11/29

    - Add new command SNAPSHOT which prints each package installed and
      its version number in the "package=version" format. Saving this
      to file will allow a Debian system to be restored to just this
      collection of packages at some later stage with the RESTORE
      command (yet to be implemented). Suggested by Dirk Eddelbuettel.
    - Update README command to report missing READMEs and missing doc
      directory. Suggested by Gintautas Miliauskas.
    - Remove remaining TABS from files to avoid mixing tabs and space.
      (Closes: #283335)

wajig (2.0.15) 2004/11/28

    - Move the main documentation out of wajig into the Survival Guide.
    - Improve README to show all documentation files found.
    - Add second argument completion for bash (Closes: #283307) 

Version 2.0.14 2004/11/16
  * Bug Fixes:
    - Handle pipes in sudo handled commands (each part of the pipe
      must be sudo). Bug pointed out by Dirk Eddelbuettel and particulary
      affected HOLD and UNHOLD when sudo was installed.

Version 2.0.13 2004/11/14
  * Bug Fixes:
    - Don't depend on sudo.  Wajig has always worked with just su.
    - Update documentation of DEPENDENTS - fixes Bug#280163
    - LARGE now sorts the list by the size of the package, and
      commifies the numbers. Sorting suggested by Gintautas Miliauskas
      in Bug#280486
    - HOLD/UNHOLD now work properly when `sudo' is not installed.
      Fixes Bug#280305 reported by Georg C. Kaindl.
    - Remove lengthy message when sudo is not installed - keep it brief
      and don't over-encourage people to use sudo since there are 
      security issues (around letting non-root passwords run commands
      as root).
    
  * New Commands:
    - README to cat the README file in /usr/share/doc. Suggested 
      by Gintautas Miliauskas in Bug#280486.
    - AUTO-ALTS to change alternatives back to autoconfigure. Suggested 
      by Gintautas Miliauskas in Bug#280486.

Version 2.0.12 2004/11/06

  * Bug Fixes:
    - The apt-setup command comes from base-config now, not apt-setup.
      Pointed out by Peter Funk.
    - Replace /etc/init.d with invoke-rc.d in wajig.py.
      Fixes Bug#276538 reported by Martin Krafft.
    - Incorporate patches from Dirk Eddelbuett:
      Makefile.in and src/Makefile.in: Use ${DESTDIR}
      wajig.sh.in and gjig.sh.in: Use -O
    - DEPENDEES becomes DEPENDENTS. Fixes Bug#277186 from Reuben Thomas.

Version 2.0.11 2004/08/09

  * Bug Fixes:
    - FORCE will now work if the package has not yet been downloaded.
      (Fixes Bug#252600 reported by Francesco Potorti)
    - CHANGELOG now handles source packages with different (but specified)
      version numbers. (Fixes Bug#248549 reported by Brad Sawatzky)
    - NEWS now handles source packages with different (but specified)
      version numbers. (Similar to Bug#248549)
    - NEWS requires the lynx package - note this if it is not installed.

Version 2.0.10 2004/05/23

  * New Commands:
    - FILE-REMOVE will remove all packages listed in file, as
      suggested by Solo Turn.
    - LIST-WIDE is the same as LIST but use a wider column to avoid
      truncated names (Reported by Karl Schmidt, Suggested by Dirk
      Eddelbuett).
    - LIST-STATUS is the same as LIST but will only list first two
      columns (desired/status and non-truncated package name). Fixes
      a problem reported by Solo Turn.

Version 2.0.9 2004/05/04

  * New command SEARCH-APT will search for well connected
    Debian archives.


wajig/bash_completion.d/wajig.bash
_have grep-status && {
_comp_dpkg_installed_packages()
{
    grep-status -P -e "^$1" -a -FStatus 'install ok installed' -n -s Package
}
} || {
_comp_dpkg_installed_packages()
{
    command grep -A 1 "Package: $1" /var/lib/dpkg/status | \
        command grep -B 1 -Ee "ok installed|half-installed|unpacked| \
            half-configured" \
            -Ee "^Essential: yes" | \
        command grep "Package: $1" | cut -d\  -f2
}
}

_have grep-status && {
_comp_dpkg_hold_packages()
{
    grep-status -P -e "^$1" -a -FStatus 'hold' -n -s Package
}
} || {
_comp_dpkg_hold_packages()
{
    command grep -B 2 'hold' /var/lib/dpkg/status | \
        command grep "Package: $1" | cut -d\  -f2
}
}

_have wajig &&
_wajig()
{
    local cur dashoptions prev special i

    COMPREPLY=()
    _get_comp_words_by_ref cur prev

    dashoptions='-h --help -V --version'

    for (( i=0; i < ${#COMP_WORDS[@]}-1; i++ )); do
        if [[ ${COMP_WORDS[i]} == \
         @(addcdrom|addrepo|aptlog|autoalts|autoclean|autodownload|autoremove|build|\
builddeps|changelog|clean|commands|contents|dailyupgrade|dependents|describe|describenew|\
distupgrade|download|editsources|extract|fixconfigure|fixinstall|fixmissing|force|hold|\
info|init|install|installsuggested|integrity|large|lastupdate|listall|listalternatives|listcache|\
listdaemons|listfiles|listhold|listinstalled|listlog|listnames|listpackages|listscripts|listsection|\
listsections|liststatus|localupgrade|madison|move|new|newdetail|news|nonfree|orphans|passwords|policy|\
purge|purgeorphans|purgeremoved|rbuilddeps|readme|reboot|recdownload|recommended|reconfigure|\
reinstall|reload|remove|removeorphans|repackage|reportbug|repos|restart|rmrepo|rpm2deb|rpminstall|search|\
searchapt|show|sizes|snapshot|source|start|status|stop|sysinfo|tasksel|todo|toupgrade|\
tutorial|unhold|unofficial|update|updatealternatives|updatepciids|updateusbids|upgrade|upgradesecurity|\
verify|version|versions|whichpackage) ]];
         then special=${COMP_WORDS[i]}
        fi
    done

    if [[ -n "$special" ]]; then
       case $special in
           install|distupgrade|download|show|changelog|builddeps|dependents|describe|details|policy|recdownload|source)
               COMPREPLY=( $( apt-cache pkgnames $cur 2> /dev/null ) )
               if [[ "$special" == "install" ]]; then
                   _filedir
               fi
               return 0
               ;;
           purge|remove|reinstall|listinstalled|hold|news|readme|recommended|reconfigure|repackage|todo|verify)
               COMPREPLY=( $( _comp_dpkg_installed_packages "$cur" ) )
               return 0
               ;;
           reload|*start|status|stop)
               _services "$cur"
               return 0
               ;;
           unhold)
               COMPREPLY=( $( _comp_dpkg_hold_packages "$cur" ) )
               return 0
               ;;
           contents|extract|info|rpm2deb|rpminstall)
               _filedir
               ;;
       esac
    fi

    case $prev in
        # don't complete anything if these options are found
        autoclean|clean|search|upgrade|update)
            return 0
            ;;
        -S)
            _filedir
            return 0
            ;;
    esac

    if [[ "$cur" == -* ]]; then
        COMPREPLY=( $( compgen -W "$dashoptions" -- "$cur" ) )
    elif [[ -z "$special" ]]; then
        commands=(addcdrom addrepo aptlog autoalts autoclean autodownload autoremove build builddeps
                  changelog clean commands contents dailyupgrade dependents describe
                  describenew distupgrade download editsources extract fixconfigure fixinstall
                  fixmissing force hold info init install installsuggested integrity large lastupdate
                  listall listalternatives listcache listdaemons listfiles listhold listinstalled
                  listlog listnames listpackages listscripts listsection listsections liststatus
                  localupgrade madison move new newdetail news nonfree orphans passwords policy purge purgeorphans
                  purgeremoved rbuilddeps readme reboot recdownload recommended reconfigure reinstall
                  reload remove removeorphans repackage reportbug repos restart rmrepo rpm2deb rpminstall search
                  searchapt show sizes snapshot source start status stop sysinfo tasksel
                  todo toupgrade tutorial unhold unofficial update updatealternatives updatepciids
                  updateusbids upgrade upgradesecurity verify version versions whichpackage)

        local option oldNoCaseMatch=$(shopt -p nocasematch)
        shopt -s nocasematch
        COMPREPLY=( $( for command in "${commands[@]}"; do
                [[ ${command:0:${#cur}} == "$cur" ]] && printf '%s\n' $command
                done ) )
        eval "$oldNoCaseMatch" 2> /dev/null
    fi

    return 0
}
complete -F _wajig wajig

# Local variables:
# mode: shell-script
# End:
